charts/opentelemetry-collector/examples/deployment-use-existing-configMap/deployment-values.yaml

```diff
@@ -0,0 +1,26 @@
+mode: deployment
+
+resources:
+  limits:
+    cpu: 100m
+    memory: 200M
+
+useExistingConfigMap: true
+
+config: null
+
+command:
+  name: otelcol-contrib
+  extraArgs: ["--config=/conf/config.yaml"]
+
+extraVolumes:
+  - name: custom-otelcol-configmap
+    configMap:
+      name: custom-otel-collector-config
+      items:
+        - key: config
+          path: config.yaml
+      defaultMode: 420
+extraVolumeMounts:
+  - name: custom-otelcol-configmap
+    mountPath: /conf/config.yaml
```

Comment:
We tend to introduce examples for only significant functional use cases. Let's do not introduce one for this use case. If we  accumulate a lot of examples, any tiny change to the chart creates a huge diff :)

cc @TylerHelmuth

Reply:
@dmitryax If we stick with the proposals solution of relying on `command.extraArgs`, `extraVolumes`, and `extraVolumeMounts` I would support this example.  Thats a lot of extra things to manage when turning on useExistingConfigMap. 

If we switch to your configMap interface idea and include in that interface options to set the content that was previous being set in `command.extraArgs`, `extraVolumes`, and `extraVolumeMounts` like in this example then I think we could forgo the example.

Reply:
@dmitryax How about moving this example's content into an existing example dir(e.g examples/daemonset-and-deployment) instead of creating a separate `deployment-use-existing-configMap` folder?

Reply:
Ok, let's keep this example since it requires many fields to be changed